firewood/src/db.rs

```diff
@@ -729,9 +734,15 @@ mod test {
                 .collect()
         }
         async fn reopen(self) -> Self {
+            self.reopen_truncate(false).await
+        }
+        async fn replace(self) -> Self {
+            self.reopen_truncate(true).await
+        }
+        async fn reopen_truncate(self, truncate: bool) -> Self {
```

Comment:
nit: there's still an unadorned bool, but it's trivial to remove it now.

It's still just a nit so feel free to ignore.
```suggestion
            self.reopen_with_config(DbConfig::default()).await
        }
        async fn replace(self) -> Self {
              self.reopen_with_config(DbConfig::builder().truncate(true).build()).await
        }
        async fn reopen_with_config(self, dbconfig: DbConfig) -> Self {
```